WORKSPACE

```diff
@@ -16,22 +16,28 @@
 
 workspace(name = "oak")
 
-load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
```

Comment:
I also looked into using `git_repository` as it seems more ergonomic, but anecdotally most projects I saw on GitHub seem to prefer `http_archive`. Did you find any supporting arguments / examples suggesting one over the other? If nothing else, I note that using `git_repository` with a tag is not necessarily reproducible (the tag may be moved on the remote repo), also see https://github.com/bazelbuild/bazel/blob/master/tools/build_defs/repo/git.bzl#L258 . So maybe if we do want to use `git_repository` we should  at least pin to specific commits. WDYT?

Reply:
Hm, indeed I didn't realize tags can be moved. Although if the maintainer moves the tag on the project (and there should be a very good reason to do so) it's something we should ideally follow right? But this indeed doesn't help with reproducibility. Here on GitHub, referring other git repositories feels quite natural over using HTTP archives. I'm not sure what's the best practice with Bazel, but other projects often use git submodules for this purpose. On the other hand if we really care about build integrity I have to agree in that we should use the HTTP archives. What do you think?

Reply:
I don't think we would want to follow tags as they are moved, the main point here is to achieve reproducibility both between developers (e.g. you and I clone this repo, build it, and expect it to produce the same artifact) and across time (e.g. I clone this repo today and then again tomorrow and I expect to produce the same artifact). Plus Bazel will either have to assume that tags never move (in which case it will miss the change if they do) or that they may move at any point (in which case it has to check the status of the tag from the remote at each invocation of any Bazel command). Additionally, I would argue that what we want to track is the state of the code at a particular point in time, not a user-defined and not particularly meaningful "tag" string.
Finally, https://docs.bazel.build/versions/master/external.html#repository-rules suggests using `http_archive`, though I wish it explained a bit better why.

Reply:
OK, I'm doing http_archives now.